main.go

```diff
@@ -57,27 +57,43 @@ func process(typeName string, fileName string, packageName string, mode string)
 
 	var specs [][2]string
 
-	ast.Inspect(f, func(node ast.Node) bool {
-		spec, ok := node.(*ast.ValueSpec)
-		if !ok {
+	ast.Inspect(f, func(astNode ast.Node) bool {
+		node, ok := astNode.(*ast.GenDecl)
+		if !ok || node.Tok != token.CONST {
 			return true
 		}
 
-		if len(spec.Names) != 1 {
-			return false
-		}
-		// TODO: check that type matches
+		var typeFound *ast.Ident
 
-		tag, ok := "", false
-		for _, field := range strings.Fields(spec.Comment.Text()) {
-			if strings.HasPrefix(field, "json:") {
-				tag, ok = field[len("json:\""):len(field)-1], true
-				break
+		for _, astSpec := range node.Specs {
```

Comment:
yeah, something like this. but I would suspected you need tree walking here.

if it is available in direct children, thats perfect.

Reply:
At this point we already know we are in `const` or `var`, and seems like in the test cases it's available in direct children. Can you think of an example where tree walking would be needed?

Reply:
no, it was just a hunch